crates/optimism/txpool/src/maintain.rs

```diff
@@ -36,16 +47,19 @@ where
     St: Stream<Item = CanonStateNotification<N>> + Send + Unpin + 'static,
 {
     async move {
-        maintain_transaction_pool(pool, events).await;
+        maintain_transaction_pool(pool, events, supervisor_client).await;
     }
     .boxed()
 }
 
 /// Maintains the state of the transaction pool by handling new blocks and reorgs.
 ///
 /// This listens for any new blocks and reorgs and updates the transaction pool's state accordingly
-pub async fn maintain_transaction_pool<N, Pool, St>(pool: Pool, mut events: St)
-where
+pub async fn maintain_transaction_pool<N, Pool, St>(
+    pool: Pool,
+    mut events: St,
+    supervisor_client: SupervisorClient,
+) where
```

Comment:
I'd like to make this its own task, because this own is already super specific to conditionals.

we can simply roll an additional one just for interop, then we add only interop logic in there